nyc_taxis/operations/default.json

```diff
@@ -25,6 +25,47 @@
       "retry-until-success": true,
       "include-in-reporting": false
     },
+    {
+      "name": "create-snapshot-repository",
+      "operation-type": "create-snapshot-repository",
+      "repository": "{{ snapshot_repository_name | default('test-repository') }}",
+      "body": {
+        "type": "s3",
+        "settings": {
+          "bucket": "{{ s3_bucket_name | default('opensearch-snapshot') }}",
+          "region": "{{ s3_bucket_region | default('us-east-1') }}"
+        }
+      }
+    },
+    {
+      "name": "create-snapshot",
+      "operation-type": "create-snapshot",
+      "repository": "{{ snapshot_repository_name | default('test-repository') }}",
+      "snapshot": "{{ snapshot_name | default('test-snapshot') }}",
+      "body": {
+        "indices": []
```

Comment:
Should the `nyc_taxis` index be in this array?

Same question for the `restore-snapshot` operation

Reply:
According to the doc (https://www.elastic.co/guide/en/elasticsearch/reference/7.10/create-snapshot-api.html#create-snapshot-api-request-body):
> By default, a snapshot includes all data streams and indices in the cluster. If this argument is provided, the snapshot only includes the specified data streams and clusters.

There 2 reasons I didn't specify the index name in the `indices` parameter:
1. There will only be one index during the benchmark run, because there is an operation to delete all the index as the first step - https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/1f85d5c56f586e1dcc787d8093055ba69f79451c/nyc_taxis/test_procedures/default.json#L7
I don't think it's necessary to add the index name.
2. I didn't see any workloads have got hard-coded index name in the test procedure definitions, and I didn't find a predefined variable name that stands for the index name of the workload, so I'd rather not hard-coding the index name here.

Reply:
> There will only be one index during the benchmark run, because there is an operation to delete all the index as the first step

I don't think this is true, and if it is then we need to fix that as it could inadvertently delete important data.

IIRC, OSB only deletes the test index. For example, if I run SO, then run NYC_TAXIS, the NYC_TAXIS run will not delete the index created for SO. Users may have data on the cluster outside of the test index, by not specifying the test index here, we are adding to the duration of the test (and potentially skewing metrics) as the snapshot and restore will occur for all indices rather than just the NYC_TAXIS test index.

Reply:
@engechas Sorry didn't see your comment in time. Thanks for your explanation, and I will add the index name explicitly.

Reply:
You are right, based on the document (https://esrally.readthedocs.io/en/stable/track.html#delete-index), `delete-index` operation (by default) will only delete the indices that specified in the workload.